File: lib/models/rentProduct.dart
import 'dart:convert';

class RentProduct {
  
  final String category;
  final String ownerName;
  final double mobileNo;
  final String houseName;
  final double rent;
  final double deposit;
  final List<String> images;
  final String gender;
  final String furnish;
  final String wifi;
  final String date;
  final String security;
  final String parking;
  final String? id;

  RentProduct({
    required this.category,
    required this.ownerName,
    required this.mobileNo,
    required this.houseName,
    required this.rent,
    required this.deposit,
    required this.images,
    required this.gender,
    required this.furnish,
    required this.wifi,
    required this.date,
    required this.security,
    required this.parking,
     this.id
  });
    


  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'category': category,
      'ownerName': ownerName,
      'mobileNo': mobileNo,
      'houseName': houseName,
      'rent': rent,
      'deposit': deposit,
      'images': images,
      'gender': gender,
      'furnish': furnish,
      'date':date,
      'wifi': wifi,
      'security': security,
      'parking': parking,
      'id': id,
    };
  }

  factory RentProduct.fromMap(Map<String, dynamic> map) {
    return RentProduct(
      category: map['category'] ?? '',
      ownerName: map['ownerName'] ?? '',
      mobileNo: map['mobileNo']?.toDouble() ?? 0.0,
      houseName: map['houseName'] ?? '',
      rent: map['rent']?.toDouble() ?? 0.0,
      deposit: map['deposit']?.toDouble() ?? 0.0,
      images: List<String>.from((map['images'])),
      gender: map['gender'] ?? '',
      date: map['date'] ?? '',
      furnish: map['furnish'] ?? '',
      wifi: map['wifi'] ?? '',
      security: map['security'] ?? '',
      parking: map['parking'] ?? '',
      id: map['_id'] ,
    );
  }

  String toJson() => json.encode(toMap());

  factory RentProduct.fromJson(String source) => RentProduct.fromMap(json.decode(source) as Map<String, dynamic>);
}

File: server/models/rentProduct.js
const mongoose = require("mongoose");


const rentProductSchema = mongoose.Schema({
  ownerName: {
    type: String,
    // required: true,
    trim: true,
  },

  mobileNo: {
    // required: true,
    trim: true,
    type:Number,
  },

  houseName:{
    type: String,
    // required:true,
    trim:true,
  },

  rent: {
    type: Number,
    // required: true,
    trim: true,
  },

  
  deposit: {
    type: Number,
    // required: true,
    trim: true,
  },

  wifi: {
    type: String,
    // required: true,
    trim: true,
  },


  security: {
    type: String,
    // required: true,
    trim: true,
  },


  parking: {
    type: String,
    // required: true,
    trim: true,
  },

  furnish: {
    type: String,
    // required: true,
    trim: true,
  },


  gender: {
    type: String,
    // required: true,
    trim: true,
  },

  date: {
    type: String,
    // required: true,
    trim: true,
  },
  
  images: [
    {
      type: String,
      required: true,
    },
  ],
  
  category: {
    type: String,
    required: true,
  },
//   ratings: [ratingSchema],
});

const RentProduct = mongoose.model("RentProduct", rentProductSchema);
module.exports = { RentProduct, rentProductSchema };